doc/api_review/core/keyIsBelow.md

```diff
@@ -83,12 +83,12 @@
 ### Tests
 
 - [ ] Function code is fully covered by tests  
-       - [ ] Line 214  
+       - [x] Line 214  
        - [ ] Line 227  
-       - [ ] Line 256
+       - [x] Line 256
 - [ ] All possible error states are covered by tests  
-       - [ ] 214 seems to be tested  
-       - [ ] 256 seems to be tested  
+       - [x] 214 seems to be tested  
+       - [x] 256 seems to be tested  
        - [ ] add test for cascading key vs root key
```

Comment:
How would root Keys compare to cascading Keys exactly? Would it even make sense to check them via `keyIsBelow` . I'm not sure I understand 100%.

Reply:
> How would root Keys compare to cascading Keys exactly?

Unfortunately, we do not have consistent behavior across different methods of comparison, keyIsBelow* or ksCut.

My current viewpoint is:

1. keys of different namespaces are not comparable (cannot be the same or below each other)
2. with the exception of the cascading namespace: keys of other namespaces can be equal or below a key of cascading namespace but not the other way round

E.g. `user:/something` is below `/` but `/something` is not below `user:`.

Reply:
Currently it seems like `/something` is below `user:` according to `keyIsBelow()`. At least the test I wrote for it fails: https://github.com/ElektraInitiative/libelektra/pull/3853/files#diff-f4ef5c85b1a0e34ed5f0241bf69ded1bf680e5ebc7e9a0bd343472a74f6d38d9R193